### ISSUE TITLE
reorderable doesn't work with demo/basic.html

### ISSUE THREAD

**Issue by BigFatDog** [BigFatDog/parcoords-es#27]
@timelyportfolio contributed this bug in #15 

Thank @timelyportfolio for also providing an online example that reproduces this issue.

> @BigFatDog, not sure if this is related or should be a separate issue, but on reorder of axes (reproducible [example](https://beta.observablehq.com/@timelyportfolio/simple-example-from-parcoord-es)), I get

> ![image](https://user-images.githubusercontent.com/837910/41851531-ce11d554-784d-11e8-9422-f4b2fe62a4c4.png)

> Thanks so much for your ambitious upgrade of `parallel-coordinates`.

**Comment by timelyportfolio** [comment]
@BigFatDog, I was able to look a little more closely and it appears `entries` [here](https://github.com/BigFatDog/parcoords-es/blob/5959a719ec0447ce6864712ebfabe881ac65b85c/src/util/colorPath.js#L45) does not return correct order of dimensions.

**Comment by timelyportfolio** [comment]
@BigFatDog I just added an array of objects data source to the [example](https://beta.observablehq.com/@timelyportfolio/simple-example-from-parcoord-es) and this works as expected, so the issue is only with array of arrays data.

**Comment by BigFatDog** [comment]
@timelyportfolio I've debugged for a while, too. I've made some progress but still cannot fix this bug yet.

According to my current observation, the root cause may reside in numeric indexes (when a matrix data is loaded and index is assigned as dimension title.).

When user reorders dimensions, indexes are re-calculated for each dimension and each dimension then re-assigned to parcoords state [here](https://github.com/BigFatDog/parcoords-es/blob/1b510710753627fd97f5d555119fd1802f23f8ef/src/api/sortDimensions.js#L8)

When `key` is numeric, `config.dimensions[p]` may assign a wrong dimension.

**Comment by BigFatDog** [comment]
I've checked the code that you've mentioned. I need more time than I expected on this bug.

**Comment by BigFatDog** [comment]
[v2.1.5](https://github.com/BigFatDog/parcoords-es/releases/tag/v2.1.5) has been released with the bug-fix

**Comment by timelyportfolio** [comment]
@BigFatDog, thank so much for this.  I just updated the example and both array of arrays and array of objects worked as expected.  Love going back to old school `Object.keys` :)  I hope to do some more research on this to figure out how to handle with `d3-entries` for reference.  Thanks again!!!

**Comment by BigFatDog** [comment]
@timelyportfolio You're welcome! Your observations (based on digging into the code) help me resolve this problem.

In fact `entires` of `d3-collection` is fine. I just tried the following code and it works:
```
import { entries } from 'd3-collections';
...
  entries(config.dimensions)
    .map(p => [
      position(p.key),
      d[p.key] === undefined
        ? getNullPosition(config)
        : config.dimensions[p.key].yscale(d[p.key]),
    ])
    .sort((a, b) => a[0] - b[0])
    .forEach((p, i) => {
      i === 0 ? ctx.moveTo(p[0], p[1]) : ctx.lineTo(p[0], p[1]);
    });
```
I agree with you that dimension order is incorrect with this issue. I've tried to fix dimension orders but failed. My approach is to sort the x-y coordination when a line is drawn:
```
[x1, y0],[x1, y1], [x2, y2], [x3, y3]....[xn, yn]
//canvas context 
ctx.moveTo(x0, y0);
ctx.lineTo(x1, y1);
ctx.lineTo(x2, y2);
```
Because each line path is drawn from left to right, the correct situation should be always:
```
x0 < x1 < x2 < x3 .... xn
```
when dimension order is incorrect and line path is messed up, the data is:
```
x1 > x2
```
Therefore, sorting the x coordination of matrix fixes the problem. (It is not a direct fix but I love it. 😄 )

The last thing is, I'm trying to deprecate `lodash` and `d3-collection` utils in favor of `ES6` native functions, both in my work and open-source code. On one hand, `ES6` is good, on the other hand, I feel those 'old school' code is cleaner and easier to read.
